### ISSUE TITLE
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float' when a light device is unavailable

### ISSUE THREAD

**Issue by robinhood-code** [make-all/tuya-local#1738]
This form is for reporting bugs.  If you have encountered a bug, you should
generally have some log messages or device diagnostics to show something is
going wrong.  If your problem is more of a question or a request for help,
then [Discussions](https://github.com/make-all/tuya-local/discussions) is
probably a more suitable venue.

**Describe the bug**

When a light device is unavailable and brightness setting is changed (via automation), the following error will be thrown. This seems to break the receive loop as well so Tuya Local cannot try reconnecting to the device:

```Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:218
integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: March 11, 2024 at 10:32:44 PM (1 occurrences)
Last logged: March 11, 2024 at 10:32:44 PM

Front Garden Bed receive loop terminated by exception unsupported operand type(s) for -: 'NoneType' and 'float'
Traceback (most recent call last):
  File "/config/custom_components/tuya_local/device.py", line 218, in receive_loop
    entity.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 992, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1113, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1052, in __async_calculate_state
    attr.update(self.state_attributes or {})
                ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 1202, in state_attributes
    data[ATTR_BRIGHTNESS] = self.brightness
                            ^^^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/light.py", line 149, in brightness
    return self._white_brightness
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/light.py", line 157, in _white_brightness
    val = color_util.value_to_brightness(r, val)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/color.py", line 782, in value_to_brightness
    max(1, round(scale_to_ranged_value(low_high_range, (1, 255), value))),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/scaling.py", line 50, in scale_to_ranged_value
    return (value - source_offset) * (
            ~~~~~~^~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'
```

<img width="1045" alt="image" src="https://github.com/make-all/tuya-local/assets/6055272/0136d2ea-fd5a-408e-bea8-0c7d7ea16ccf">

**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:

1. Make a light device unavailable
2. run automation to change the brightness of the device
3. error will be thrown

**Expected behavior**

1. None is checked / error is caught and doesn't prevent TL to try reconnecting to the device

**Additional context**

If the bug involves a device, then please include device diagnostics from
Settings / Devices & Services / Tuya Local (your device) / "1 device"

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.

**Comment by hellboy-wien** [comment]
Hello, i dont know, but i have FAN (Windcalm) after some Updates the device stays Unavailable:
Maybe it helps in that issue

Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.tuya_local.device
Quelle: custom_components/tuya_local/device.py:218
Integration: Tuya Local (Dokumentation, Probleme)
Erstmals aufgetreten: 13. März 2024 um 16:41:20 (1 Vorkommnisse)
Zuletzt protokolliert: 13. März 2024 um 16:41:20

Deckenventilator receive loop terminated by exception unsupported operand type(s) for -: 'NoneType' and 'float'
Traceback (most recent call last):
  File "/config/custom_components/tuya_local/device.py", line 218, in receive_loop
    entity.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 992, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1113, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1052, in __async_calculate_state
    attr.update(self.state_attributes or {})
                ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/fan/__init__.py", line 426, in state_attributes
    data[ATTR_PERCENTAGE] = self.percentage
                            ^^^^^^^^^^^^^^^
  File "/config/custom_components/tuya_local/fan.py", line 121, in percentage
    val = ranged_value_to_percentage(r, val)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/percentage.py", line 79, in ranged_value_to_percentage
    return scale_ranged_value_to_int_range(low_high_range, (1, 100), value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/scaling.py", line 23, in scale_ranged_value_to_int_range
    (value - source_offset)
     ~~~~~~^~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'

**Comment by make-all** [comment]
The fan issue was already fixed under #1513
